src/handlers/courtesy.ts

```diff
@@ -121,38 +138,47 @@ export const courtesyPutHandler = (client: Client<"SubscriptionKey">) => (
               }),
             err => {
               log.error(`upsertServiceActivation responded with: ${err}`);
-              return ResponseErrorInternal(
-                "Could not update the service activation on functions-services"
+              return PNResponseErrorInternal(
+                "Not found",
+                "Could not update the service activation on functions-services",
+                []
               );
             }
           )
         ),
         //
-        TE.mapLeft(err => ResponseErrorInternal(err.kind))
+        TE.mapLeft(err =>
+          PNResponseErrorInternal(err.kind, "COSMOS query error", [])
+        )
       );
     }),
     // The client reject the promise only if something went wrong, like a JSON.parse of the output
     // the response could contain HTTP error codes so it is necessary to filter out the response
     TE.map(
       flow(
         E.map(response => {
+          // eslint-disable-next-line sonarjs/no-small-switch
           switch (response.status) {
             case 200:
               return ResponseSuccessNoContent();
             default:
               log.error(
                 `upsertServiceActivation returned ${response.status}: ${response.value}`
               );
-              return ResponseErrorInternal(
-                "Could not update the service activation on functions-services"
+              return PNResponseErrorInternal(
+                "Update error",
+                "Could not update the service activation on functions-services",
+                []
               );
           }
```

Comment:
I think you could replace this switch with something like 
```ts
E.chain(E.fromPredicate((r) => r.status === 200), () => /*default case*/))
```
In this way I think it is cleaner and you don't need the comment to skip the eslint rule